File: stack/py/stack.py
class Stack:
    def __init__(self):
        self.stack = []

    def push(self, n):
        self.stack.append(n)

    def pop(self):
        return self.stack.pop()
    
    def peek(self):
        return self.stack[-1] if self.stack else None

    def is_empty(self):
        return not self.stack
    
    def size(self):
        return len(self.stack)

File: stack/ts/stack.ts
export class Stack<T> {
    private stack: T[] = [];

    push(n: T): void {
        this.stack.push(n);
    }

    pop(): T | undefined {
        return this.stack.pop();
    }

    peek(): T | undefined {
        return this.stack[this.stack.length - 1]
    }

    is_empty(): boolean {
        return this.stack.length === 0;
    }

    size(): number {
        return this.stack.length;
    }
}
